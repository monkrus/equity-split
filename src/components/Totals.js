import React from 'react'
import { formatCurrency } from './Utils'

export default function Totals({totals}) {

  if(Object.keys(totals).length !== 0) {
    return(
      <tr>
          <td>{'Totals'}</td>
          <td>{''}</td>
          <td>{formatCurrency(totals.totalShare)}</td>
          <td>{totals.totalSharePercent.toFixed(2)}</td>
          <td>{totals.totalFixedShare.toFixed(2)}</td>
          <td>{formatCurrency(totals.totalNonCash)}</td>
          <td>{formatCurrency(totals.totalInvestedCash)}</td>
          <td>{''}</td>
          <td>{totals.totalDays}</td>
          <td>{''}</td>
          <td>{totals.totalWorkedHours}</td>
          <td>{''}</td>
          <td>{formatCurrency(totals.totalSalary)}</td>
      </tr>
    );
  }else
    return false
}
