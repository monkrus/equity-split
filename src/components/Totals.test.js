import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Totals from './Totals'

describe(Totals, () => {
  const data = {totalShare:0, totalSharePercent:0, totalFixedShare:0,
    totalNonCash:0, totalInvestedCash:0, totalDays:0, totalWorkedHours:0, totalSalary:0};
  const label = 'Totals';
  const component = shallow(
    <Totals totals={data} />
  );

  //Generic Snapshot Test
  it('renders and matches snapshot', () => {
    const component = renderer.create(
      <Totals totals={data} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  //Content Test
  it('contains totals label', () => {
    expect(component.text()).toContain(label)
  });
})
