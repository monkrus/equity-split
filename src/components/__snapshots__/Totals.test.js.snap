exports[`function Totals({ totals }) {

  if (Object.keys(totals).length !== 0) {
    return (
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'tr', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this },
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this }, 'Totals'),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this }, ''),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this }, /* istanbul ignore next */(0, _Utils.formatCurrency)(totals.totalShare)),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 12 }, __self: this }, /* istanbul ignore next */(0, _Utils.formatLocalCurrency)(totals.totalShareNumbers)),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 13 }, __self: this }, totals.totalSharePercent.toFixed(2)),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: this }, totals.totalFixedShare.toFixed(2)),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this }, /* istanbul ignore next */(0, _Utils.formatCurrency)(totals.totalNonCash)),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }, /* istanbul ignore next */(0, _Utils.formatCurrency)(totals.totalInvestedCash)),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this }, ''),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this }, totals.totalDays),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this }, ''),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: this }, totals.totalWorkedHours),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }, ''),
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this }, /* istanbul ignore next */(0, _Utils.formatCurrency)(totals.totalSalary))));


  } else
  return false;
} renders and matches snapshot 1`] = `
<tr>
  <td>
    Totals
  </td>
  <td>
    
  </td>
  <td>
    
  </td>
  <td>
    
  </td>
  <td>
    0.00
  </td>
  <td>
    0.00
  </td>
  <td>
    
  </td>
  <td>
    
  </td>
  <td>
    
  </td>
  <td>
    0
  </td>
  <td>
    
  </td>
  <td>
    0
  </td>
  <td>
    
  </td>
  <td>
    
  </td>
</tr>
`;
